### PR TITLE
Save best solution rather than last one in sciml_train

### PR DESCRIPTION
**Pull request by ChrisDeGrendele**
This PR saves the best solution from sciml_train rather than the last one produced. This will be helpful for things like stochastic loss functions. This is only first of three sciml_train, I was hoping to see if I am on the right track.
Fixes Issue #170

### CODE REVIEW

File: src/train.jl [hunk 1]
```diff
@@ -86,7 +86,9 @@ function sciml_train(loss, _θ, opt, _data = DEFAULT_DATA;
 
   t0 = time()
 
-  local x
+  local x, min_err
+  min_err = 999999 #dummy variables
```
Comment (ChrisRackauckas) [inline src/train.jl]: typemax(eltype(_θ))

File: src/train.jl [hunk 2]
```diff
@@ -102,6 +104,15 @@ function sciml_train(loss, _θ, opt, _data = DEFAULT_DATA;
       msg = @sprintf("loss: %.3g", x[1])
       progress && ProgressLogging.@logprogress msg i/maxiters
       update!(opt, ps, gs)
+
+      if x < min_err  #found a better solution
```
Comment (ChrisRackauckas) [inline src/train.jl]: we should have it be an option to do this.

**Comment by titus-pinta** [comment]
Sorry if this is a mistake, or if I do not post in the correct place, but there seems to be a mistake. If I return more than 1 variable from my loss function in order to use the callback, the function first(x) on line 130 of train.jl will try to compare a pair of objects and will throw an error. I think that replacing it with first(x[1]) should solve the problem.
Sorry if I miss something obvious here!

**Comment by ChrisDeGrendele** [comment]
@titus-pinta Hey, good catch. I was just able to reproduce your error, so I can have a fix pushed in a bit. Thanks!

**Pull request merged**